src/basins/edgetracking.jl

```diff
@@ -0,0 +1,194 @@
+* `tmax = Inf`: maximum integration time of parallel trajectories until re-bisection 
+* `dt = 0.01`: integration time step
+* `ϵ_mapper = 0.1`: `ϵ` parameter in [`AttractorsViaProximity`](@ref)
+* `verbose = false`: if true, prints info while running
+* `output_level = 2`: controls what to return (see below)
+* `kwargs...`: additional keyword arguments to be passed to `AttractorsViaProximity`
+
+## Description
+The edge tracking algorithm is a simple numerical method to find the *edge state* or
+(possibly chaotic) saddle on the boundary between two basins of attraction. It is first
+described by Skufca et al. (2006)[^1], and has been applied to e.g. the laminar-turbulent
+boundary in plane Couette flow[^2] as well as Melancholia states in a climate model
+separating the snowball and warm climate states[^3]. Relying only on forward integration of
+the system, it works even in high-dimensional systems with complicated basin boundary
+structures.
+
+The algorithm consists of two main steps: bisection and tracking. First, it iteratively 
+bisects along a
+straight line in state space between the intial states `u1` and `u2` to find the basin
+boundary. The bisection stops when the two updated states are less than `eps1` (in
+Euclidean distance) apart from each other. Next, a `ParallelDynamicalSystem` is initialized
+from these two updated states and integrated forward until the two trajectories diverge
+from each other by more than `eps2` (Euclidean distance). The two final states of the
+parallel integration are then used as new states `u1` and `u2` for a new bisection, and 
+so on, until a stopping criterion is fulfilled.
+
+## Output 
+Output can be controlled via the `output_level` argument.
+* `output_level = 0`: returns only the final state on the edge
+* `output_level = 1`: returns the track along the edge as a vector of states
+* `output_level = 2`: returns `[track, bisect1, bisect2]`, where `track` is the track along  
+  the edge and `bisect1`, `bisect2` are vectors containing the bisection points on either  
+  side of the basin boundary
+
+!!! warning
+    May behave erroneously when the DiffEq solver of `ds` is set to `alg = SimpleATsit5()`,
+    which is the default solver in DynamicalSystems. The recommended solver is `Vern9()`.
+
+## References
+[^1]: [Skufca et al. (2006)](https://journals.aps.org/prl/pdf/10.1103/PhysRevLett.96.174101?casa_token=RUn26KnFdNEAAAAA%3AoXsTlmEWVMkEYbOtR-j2PH2vYOOPOy1a2R_37ncnf4gsiHp6GR66M-IBpzXocLoMQC_oHhk8MIFRa_8)
+[^2]: [Schneider et al. (2008)](https://journals.aps.org/pre/pdf/10.1103/PhysRevE.78.037301?casa_token=mLQmTv_cBGUAAAAA%3AVKnQs290sq1MNm-5k8hW7nJeLtVX54I7l-SEGol_HUPSCwziPi-EGDE8ucrDiVMIXZGUbzzam8benFw)
+[^3]: [Luarini and Bodai (2017)](https://iopscience.iop.org/article/10.1088/1361-6544/aa6b11)
+"""
+function edgetracking(ds::DynamicalSystem, u1, u2, attractors::Dict;
+    eps1=1e-7,
+    eps2=1e-6,
+    maxiter=100,
+    abstol=0.0,
+    tmax=Inf,
+    dt=0.01,
+    ϵ_mapper=0.1,
```

Comment:
0.1 is a rather arbitrary value for this distance. Why do you even need to set one? Isn't the default that `AttractorsViaProximity` creates a much better choice overall?

Reply:
I agree that the default of `AttractorsViaProximity` creates a good overall choice for `ϵ`, but in some cases the user may want to set it explicitly. Since `ϵ` is an optional positional argument rather than a keyword argument, I did not know how it could be passed to `AttractorsViaProximity` without creating an explicit kwarg.

I now did this - is this a good solution?
```julia
function edgetracking(ds::DynamicalSystem, attractors::Dict;
    ...
    ϵ_mapper=nothing,
    kwargs...)
    
    ...

    if ϵ_mapper == nothing
        mapper = AttractorsViaProximity(ds, attractors; kwargs...)
    else
        mapper = AttractorsViaProximity(ds, attractors, ϵ_mapper; kwargs...)
    end

    ...
```

Reply:
Simply check the source code of `AttractorsViaProximity` and get the same syntax that it uses for the default `ε` value. Make this syntax the keyword, and then just pass the keyword.